boulderlog/Program.cs

```diff
@@ -141,13 +141,18 @@ public static void Main(string[] args)
 
             app.UseCors();
             app.UseResponseCaching();
+            app.UseAuthentication(); //To ensure that the authentication middleware is used.
             app.UseAuthorization();
 
             app.MapControllerRoute(
                 name: "default",
                 pattern: "{controller=Home}/{action=Index}/{id?}");//.RequireRateLimiting(rateLimitOptions.Policy);
             app.MapRazorPages();//.RequireRateLimiting(rateLimitOptions.Policy);
-
+            app.MapFallback(context =>
+           {
+             context.Response.Redirect("Identity/Account/Login"); //fallback route redirects to Identity/Account/Login whenever an invalid URL is accessed 
```

Comment:
This change will redirect all unknown routes to the login page. Ideally anonymous routes should redirect to the home page and only routes that require authentication should redirect to the login page.